src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/UsePropertiesWhereAppropriateTests.cs

```diff
@@ -117,6 +117,11 @@ public IEnumerator GetEnumerator()
         return null;
     }
 
+    public int GetPinnableReference() // Ideally, the method return type must be `ref`.
```

Comment:
Let me know if you want this to report a diagnostic, and only ref-returns doesn't.

Reply:
Does pattern-based `fixed` allow `GetPinnableReference` to return a pointer type?

I believe the exclusion should only apply to methods that match the pattern by filtering to `ref` and `ref readonly` returns.

Reply:
According to the doc it's even more restrictive than this: https://docs.microsoft.com/dotnet/csharp/language-reference/keywords/fixed-statement

Reply:
The `ref` check seems easy and catches most cases. I don't think we need to check the unmanaged type restriction.